behaviors/behaviors.py

```diff
@@ -98,6 +100,40 @@ def released(self):
         return self.release_date and self.release_date < timezone.now()
 
 
+class Slugged(models.Model):
+    """
+    An abstract behavior representing adding a unique slug to a model
+    based on the slug_source property.
+    """
+    slug = models.SlugField(max_length=255, unique=True)
+
+    class Meta:
+        abstract = True
+
+    def save(self, *args, **kwargs):
+        if not self.slug:
```

Comment:
Sorry for the delay on this, I was moving and have had every night booked up cleaning, packing, moving, unpacking. Should be merged by this weekend.